### PR TITLE
cpu/mips: Remove all mips

### PR DESCRIPTION
**Pull request by MrKevinWeiss**
### Contribution description

Starting the remove mips CPU that was originally done by @jcarrano 

### Testing procedure

Green CI and read it.

### CODE REVIEW

File: doc/doxygen/src/flashing.md [hunk 1]
```diff
@@ -37,7 +37,7 @@ listed as supported for MCU family `BAR`.
 This table will use the value to pass to `PROGRAMMER=<value>` as title, rather
 than the official spelling of the programmer.
 
-MCU Family     | `adafruit-nrfutil` | `avrdude` | `bmp` | `bossa` | `cc2538-bsl`| `cpy2remed` | `dfu-util` | `edbg` | `elf2uf2` | `esptool` | `goodfet` | `jlink` | `lpc2k_pgm` | `mspdebug` | `nrfutil` | `openocd` | `pic32prog` | `pyocd` | `robotis-loader` | `stm32flash` | `stm32loader` | `uf2conv` | `uniflash`
+MCU Family     | `adafruit-nrfutil` | `avrdude` | `bmp` | `bossa` | `cc2538-bsl`| `cpy2remed` | `dfu-util` | `edbg` | `elf2uf2` | `esptool` | `goodfet` | `jlink` | `lpc2k_pgm` | `mspdebug` | `nrfutil` | `openocd` | `pyocd` | `robotis-loader` | `stm32flash` | `stm32loader` | `uf2conv` | `uniflash`
```
Comment (maribu) [inline doc/doxygen/src/flashing.md:40]: Removing one column caption would need to remove all of the column contents.

Maybe it really was a poor idea to include single-platform flash tools in this matrix. May I should have created a matrix for multi-platform flash tools and listed the single-platform flashers separately.

**Review by maribu — COMMENTED** [review]
Thx for tackling this. It really is impressive to see what level of effort the MIPS toolchain caused and how little it is used

File: doc/doxygen/src/flashing.md (hunk 1 above)
Comment (maribu) [reply doc/doxygen/src/flashing.md:40]: I split out the single-purpose tools in PR https://github.com/RIOT-OS/RIOT/pull/18577

**Review by maribu — APPROVED** [review]
ACK. This needs a rebase now (and adaption of the changes to the flashing doc), other than that it should be good to go

**Comment by maribu** [comment]
There seems to still be some files left. The CI is listing `cpu/mips32r2_common/include/cpu.h` below "Unchanged files with check annotations"

**Comment by MrKevinWeiss** [comment]
Rebased and updated lost and found!

**Comment by MrKevinWeiss** [comment]
Thanks, the codebase is getting cleaner and cleaner!

**Pull request merged**